### DISCUSSION TITLE
HASP Keeps resetting or losing network connection.

### DISCUSSION THREAD

**Discussion by eboston** [HASwitchPlate/HASPone#128] [Q&A]
I have a HASP unit on my desk that I am about to throw into a shredder.  It keeps losing network connection or resets (probably because of the network connection).  None of my other units do this.  The only difference with this one is that it has the latest firmware.  I'm also thinking it could be the ESP module is not that great.  I can't even get it online long enough to look at which version of firmware is actually on it.  It is the only unit I have upgraded to the blueprints. 

Here is the debug output come through the USB port.  It keeps cycling showing the same thing each time.

[+671.504s] MQTT OUT: 'hasp/desktop/state/page' : '2'
[+679.977s] Reconnecting to WiFi network...
[+684.934s] MQTT: not connected, connecting.
[+684.939s] MQTT: Configuring MQTT connection without TLS.
[+684.945s] HMI OUT: page 0
[+684.948s] HMI OUT: 'p[0].b[1].font=6'
[+685.130s] HMI IN: [5 bytes]: 0x66 0x00 0xff 0xff 0xff
[+685.135s] HMI IN: [sendme Page] '0'
[+685.140s] HMI OUT: 'p[0].b[1].txt="WiFi Connected!\rë Oxford\rIP: 192.168.1.56\r\rMQTT Connecting:\r 192.168.1.200"'
[+685.152s] MQTT: Attempting connection to broker 192.168.1.200 on port 1883 with TLS disabled as clientID desktop-a4cf12d5ca9d
[+685.194s] HMI OUT: 'p[0].b[1].txt="WiFi Connected!\rë Oxford\rIP: 192.168.1.56\r\rMQTT Connected:\r 192.168.1.200"'
[+685.206s] MQTT: connected
[+685.214s] MQTT: subscribed to hasp/desktop/command/#
[+685.223s] MQTT IN: 'hasp/desktop/command/page' : '2'
[+685.228s] HMI OUT: page 2
[+685.432s] MQTT: subscribed to hasp/plates/command/#
[+685.441s] MQTT: subscribed to hasp/desktop/light/switch/#
[+685.452s] MQTT: subscribed to hasp/desktop/brightness/set/#
[+685.469s] MQTT OUT: 'homeassistant/light/desktop/config' : '{"name":"desktop backlight","command_topic":"hasp/desktop/light/switch","state_topic":"hasp/desktop/light/state","brightness_state_topic":"hasp/desktop/brightness/state","brightness_command_topic":"hasp/desktop/brightness/set","availability_topic":"hasp/desktop/status","brightness_scale":100,"unique_id":"desktop-a4cf12d5ca9d-backlight","payload_on":"ON","payload_off":"OFF","payload_available":"ON","payload_not_available":"OFF","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.02}}'
[+685.536s] MQTT OUT: 'homeassistant/sensor/desktop/config' : '{"name":"desktop sensor","json_attributes_topic":"hasp/desktop/sensor","state_topic":"hasp/desktop/status","unique_id":"desktop-a4cf12d5ca9d-sensor","icon":"mdi:cellphone-text","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.02}}'
[+685.581s] MQTT OUT: 'homeassistant/number/desktop/config' : '{"name":"desktop active page","command_topic":"hasp/desktop/command/page","state_topic":"hasp/desktop/state/page","retain":true,"optimistic":true,"icon":"mdi:page-next-outline","unique_id":"desktop-a4cf12d5ca9d-page","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.02}}'
[+685.628s] MQTT OUT: 'hasp/desktop/alwayson' : 'ON'
[+685.638s] MQTT OUT: 'homeassistant/light/desktop/selectedforegroundcolor/config' : '{"name":"desktop selected foreground color","command_topic":"hasp/desktop/light/selectedforegroundcolor/switch","state_topic":"hasp/desktop/alwayson","rgb_command_topic":"hasp/desktop/light/selectedforegroundcolor/rgb","rgb_command_template":"{{(red|bitwise_and(248)*256)+(green|bitwise_and(252)*8)+(blue|bitwi▒e_and(248)/8)|int }}","retain":true,"unique_id":"desktop-a4cf12d5ca9d-selectedforegroundcolor","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.02}}'
[+685.708s] MQTT OUT: 'homeassistant/light/desktop/selectedbackgroundcolor/config' : '{"name":"desktop selected background color","command_topic":"hasp/desktop/light/selectedbackgroundcolor/switch","state_topic":"hasp/desktop/alwayson","rgb_command_topic":"hasp/desktop/light/selectedbackgroundcolor/rgb","rgb_command_template":"{{(red|bitwise_and(248)*256)+(green|bitwise_and(252)*8)+(blue|bitwise_and(248)/8)|int }}","retain":true,"unique_id":"desktop-a4cf12d5ca9d-selectedbackgroundcolor","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.02}}'
[+685.776s] MQTT OUT: 'homeassistant/light/desktop/unselectedforegroundcolor/config' : '{"name":"desktop unselected foreground color","command_topic":"hasp/desktop/light/unselectedforegroundcolor/switch","state_topic":"hasp/desktop/alwayson","rgb_command_topic":"hasp/desktop/light/unselectedforegroundcolor/rgb","rgb_command_template":"{{(red|bitwise_and(248)*256)+(green|bitwise_and(252)*8)+(blue|bitwise_and(248)/8)|int }}","retain":true,"unique_id":"desktop-a4cf12d5ca9d-unselectedforegroundcolor","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.02}}'
[+685.849s] MQTT OUT: 'homeassistant/light/desktop/unselectedbackgroundcolor/config' : '{"name":"desktop unselected background color","command_topic":"hasp/desktop/light/unselectedbackgroundcolor/switch","state_topic":"hasp/desktop/alwayson","rgb_command_topic":"hasp/desktop/light/unselectedbackgroundcolor/rgb","rgb_command_template":"{{(red|bitwise_and(248)*256)+(green|bitwise_and(252)*8)+(blue|bitwise_and(248)/8)|int }}","retain":true,"unique_id":"desktop-a4cf12d5ca9d-unselectedbackgroundcolor","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.02}}'
[+685.913s] MQTT OUT: 'hasp/desktop/light/state' : 'ON'
[+685.926s] MQTT OUT: 'hasp/desktop/brightness/state' : 100
[+685.938s] MQTT: subscribed to hasp/desktop/status
[+685.947s] HMI IN: [5 bytes]: 0x66 0x02 0xff 0xff 0xff
[+685.952s] HMI IN: [sendme Page] '2'
[+685.956s] M▒TT OUT: 'hasp/d▒sktop/state/json' : '{"event":"page","value":2}'
[+685.965s] MQTT OUT: 'hasp/desktop/state/page' : '2'
[+694.406s] Reconnecting to WiFi networ▒...

**Comment by SteveRMann** [comment]
What happens if you move the HASP closer to the router?
